be/src/exec/vectorized/aggregator.h

```diff
@@ -139,11 +150,74 @@ static const int STREAMING_HT_MIN_REDUCTION_SIZE =
 
 using AggregatorPtr = std::shared_ptr<Aggregator>;
 
-// Component used to process aggregation including bloking aggregate and streaming aggregate
+struct AggregatorParams {
+    bool needs_finalize;
+    bool has_outer_join_child;
+    int64_t limit;
+    TStreamingPreaggregationMode::type streaming_preaggregation_mode;
+    TupleId intermediate_tuple_id;
+    TupleId output_tuple_id;
+    std::string sql_grouping_keys;
+    std::string sql_aggregate_functions;
+    std::vector<TExpr> conjuncts;
+    std::vector<TExpr> grouping_exprs;
+    std::vector<TExpr> aggregate_functions;
+    std::vector<TExpr> intermediate_aggr_exprs;
+
+    // streaming mv variables
+    bool is_stream_mv;
+    bool is_generate_retract;
+    bool is_testing;
+    int32_t count_agg_idx;
+};
+
+static std::shared_ptr<AggregatorParams> convert_to_aggregator_params(const TPlanNode& tnode) {
```

Comment:
move implementation into cpp files as much as possible. Because it will make our compile time more longer